game/missiongenerator/aircraft/flightgroupspawner.py

```diff
@@ -93,8 +93,7 @@ def generate_flight_at_departure(self) -> FlyingGroup[Any]:
                 group = self._generate_over_departure(name, cp)
                 return group
             elif isinstance(cp, NavalControlPoint):
-                group_name = cp.get_carrier_group_name()
-                carrier_group = self.mission.find_group(group_name)
+                carrier_group = cp.get_carrier_group()
```

Comment:
I don't think this is right. The `Group` in the mission and the `Group` in the TGO are not the same object. The `Group` in `TheaterGroundObject` is just a template for what to spawn in the mission. The thing in `Mission` is the actual group that will spawn. They probably have different IDs. Did you test that aircraft on carriers actually get assigned to the carrier in the generated mission? If so I don't understand why this works :(

Reply:
You are right, thanks for spotting this! Fixed in 2352fbfcfef7ea43625b76fae0913733e326ed2a